### PR TITLE
Changes based on my usability studies

### PR DESCRIPTION
**Pull request by acknudson**
@afmckinney @jisqyv @halatmit

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java [hunk 1]
```diff
@@ -617,23 +550,92 @@
...
+  @Description("Menu item to update the Companion to the latest version")
+  String companionUpdate();
+
+  @DefaultMessage("You must have a project open to update the Companion")
+  @Description("")
+  String companionUpdateMustHaveProject();
+
+  @DefaultMessage("Update Companion on Emulator")
+  @Description("Hard Reset the Emulator.")
+  String hardResetConnectionsMenuItem();
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: @ajcolter I'm not sure I'm happy with this wording. Hard Reset does *not* update the Companion on the Emulator. Instead it resets it to its initial as installed state (which at this point is an ancient companion). The auto-update mechanism then will prompt to update it on next connection. Maybe we should just remove this item now that we have the "Update the Companion" functionality. (I'll double check that it works on the emulator first, but it should).

Comment (ajcolter) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: @jisqyv We can use whatever wording you think makes sense (if you want to keep this option). However, Hard Reset doesn't really tell the user what the option does (I've clicked it unwittingly many times), and from what I could gather on the documentation, it seemed like it was updating on the Emulator (by first resetting). We can work this out.

**Comment by jisqyv** [comment]
@ajcolter It looks like in at least one place, and probably others you edited the "name" of a string in OdeMessages as well as the displayed text. This results in all of the non-English translations displaying the English version because translated versions of the new values do not exist. If you leave the "name" of the string the same, the old message will be displayed in the appropriate language.  Until we translate the updated text strings, it is probably better to show the old versions in someone's native language then to show them the new version in English.

File: appinventor/appengine/src/com/google/appinventor/client/Ode.java [hunk 2]
```diff
@@ -189,13 +190,13 @@
   // Remembers the current View
   static final int DESIGNER = 0;
   static final int PROJECTS = 1;
-  private static final int GALLERY = 2;
+  static final int GALLERY = 2;
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/Ode.java]: @ajcolter Why was the "private" keyword removed here?

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java [hunk 3]
```diff
@@ -349,7 +330,7 @@
   // Used in Ode.java
 
   // TODO(user): Replace with commented version once we're ready
-  @DefaultMessage("MIT App Inventor 2")
+  @DefaultMessage("MIT App Inventor")
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: @ajcolter Did we agree to this branding change?

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java [hunk 4]
```diff
@@ -460,89 +437,69 @@
...
+  //Live-test
+  @DefaultMessage("Test")
   @Description("Label of the button leading to Connect related cascade items")
   String connectTabName();
 
-  @DefaultMessage("AI Companion")
+  @DefaultMessage("Companion App over WiFi")
   @Description("Message providing details about starting the wireless connection.")
-  String AICompanionMenuItem();
+  String aiCompanionMenuItem();
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: @ajcolter This change breaks translations

Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: @ajcolter You can make this change, but then you should update all of the translation properties files as well (and we have to update all of the translations in Pootle!).

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java (hunk 3 above)
Comment (halatmit) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: I'm in favor of making this branding change.

Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: I don't have a problem with it, but we should probably separate it out into its own commit.

Comment (halatmit) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: Good idea to separate it.

Comment (ajcolter) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: Done. See new PR for this change.

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java (hunk 4 above)
Comment (ajcolter) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: I changed this one back.

**Comment by ajcolter** [comment]
@jisqyv I probably did change the "name" of a string in OdeMessages, and I created a few new strings there. Where is the Pootle server/the place where I need to make these changes (so the translations can be added)?

File: appinventor/appengine/src/com/google/appinventor/client/Ode.java (hunk 2 above)
Comment (ajcolter) [inline appinventor/appengine/src/com/google/appinventor/client/Ode.java]: I think this was due to some changes that I made in TopToolbar.java. However, I just reviewed all of my changes, and I can't find GALLERY used, so I'll put the private keyword back on.

**Comment by afmckinney** [comment]
I am actively looking at this now.

**Comment by ewpatton** [comment]
We need to think about how to break this up into smaller chunks that can be accepted, and how it interacts with Paul's work for the Youth Radio toolkits in the Tutorial panel, among others.
